src/thermo/src/4C_thermo_timint.cpp

```diff
@@ -377,11 +377,13 @@ void Thermo::TimInt::read_restart_state()
  *----------------------------------------------------------------------*/
 void Thermo::TimInt::write_runtime_output()
 {
-  runtime_output_writer_.value()->reset();
+  FOUR_C_ASSERT_ALWAYS(runtime_output_writer_.has_value(), "Runtime output not initialized.");
+
+  runtime_output_writer_.value().reset();
 
   if (runtime_output_params_.output_temperature_state)
   {
-    runtime_output_writer_.value()->append_result_data_vector_with_context(
+    runtime_output_writer_.value().append_result_data_vector_with_context(
```

Comment:
```suggestion
    runtime_output_writer_->append_result_data_vector_with_context(
```

You can just access the optional like that as you asserted that it is non-empty.

Reply:
Right, that works now :+1:.